### PR TITLE
Move images to grouchi-assets

### PR DESCRIPTION
**Pull request by nathanchica**
This pull request introduces a new utility function to standardize and centralize how image asset URLs are generated for cat profiles, replacing hardcoded paths throughout the codebase. It also adds comprehensive unit tests for this function and updates documentation to clarify image hosting.

**Asset URL Handling Improvements**

* Added `getAssetUrl` utility function in `src/utils/assets.ts` to generate jsDelivr URLs for cat images, supporting both navigation and numbered images, and handling alias formatting.
* Replaced hardcoded image paths in `src/data/cats.ts` and `routes/cats.ts` with calls to `getAssetUrl`, ensuring consistency and maintainability for asset references. [[1]](diffhunk://#diff-5d5bea80a644c6789aa02796c4502f523d3025f983614057edd67dad3672ed09L21-R24) [[2]](diffhunk://#diff-5d5bea80a644c6789aa02796c4502f523d3025f983614057edd67dad3672ed09L43-R46) [[3]](diffhunk://#diff-5f3f159fe8650c17e76afaf2909cf68fb3aabbba10e832dc509ddbc17dba28f9L15-R16) [[4]](diffhunk://#diff-5f3f159fe8650c17e76afaf2909cf68fb3aabbba10e832dc509ddbc17dba28f9L91-R92)

**Testing and Documentation**

* Added unit tests for `getAssetUrl` in `src/utils/__tests__/assets.test.ts`, covering various alias formats, type variations, and edge cases.
* Updated `README.md` to document the use of the `grouchi-assets` repository and jsDelivr for image delivery.

### CODE REVIEW

File: src/utils/assets.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export function getAssetUrl(alias: string, type: 'nav' | number): string {
+    const formattedAlias = alias.replaceAll('_', '-');
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/utils/assets.ts:2]: this isnt running in a browser, its in the express server

File: src/utils/assets.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    const filePrefix = formattedAlias === 'groucho-and-chica' ? 'grouchi' : formattedAlias;
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/utils/assets.ts:3]: maybe if there will be more cases
